lib/record_store/provider/dnsimple.rb

```diff
@@ -29,42 +30,50 @@ def add(record, zone)
       end
 
       def remove(record, zone)
-        session.delete_record(zone, record.id)
+        session.zones.delete_record(account_id, zone, record.id)
       end
 
       def update(id, record, zone)
-        session.update_record(zone, id, api_hash(record, zone))
+        session.zones.update_record(account_id, zone, id, api_hash(record, zone))
       end
 
       # returns an array of Record objects that match the records which exist in the provider
       def retrieve_current_records(zone:, stdout: $stdout)
-        session.list_records(zone).body["data"].map do |record_body|
+        session.zones.all_records(account_id, zone).data.map do |record|
+          params = parameterize_record(record)
+
           begin
-            build_from_api(record_body, zone)
+            build_from_api(params, zone)
           rescue StandardError
-            stdout.puts "Cannot build record: #{record_body}"
+            stdout.puts "Cannot build record: #{params}"
             raise
           end
         end.compact
       end
 
       # Returns an array of the zones managed by provider as strings
       def zones
-        session.zones.map(&:domain)
+        session.zones.all_zones(account_id).data.map(&:name)
       end
 
       private
 
       def session
-        @dns ||= Fog::DNS.new(session_params)
+        @dns ||= Dnsimple::Client.new(
+          base_url: secrets.fetch('base_url'),
+          access_token: secrets.fetch('api_token')
+        )
       end
 
-      def session_params
-        {
-          provider: 'DNSimple',
-          dnsimple_token: secrets.fetch('api_token'),
-          dnsimple_account: secrets.fetch('account_id'),
-        }
+      def account_id
+        @account_id ||= secrets.fetch('account_id')
+      end
+
+      def parameterize_record(record)
+        record.instance_variables.each_with_object({}) do |attr, hash|
```

Comment:
this smells a little to me... what type is `record`?

Reply:
ie. what class?

Reply:
This is just a translation from a object to a hash to avoid changing other methods. Here's the type of the object: http://www.rubydoc.info/gems/dnsimple/Dnsimple/Struct/ZoneRecord

Reply:
Just removed the translation layer and update the `build_from_api` to accommodate dnsimple objects.